x/bsonx/bsoncore/bsoncore.go

```diff
@@ -576,55 +576,54 @@ func AppendInt64(dst []byte, i64 int64) []byte { return appendi64(dst, i64) }
 // AppendInt64Element will append a BSON int64 element using key and i64 to dst
 // and return the extended buffer.
 func AppendInt64Element(dst []byte, key string, i64 int64) []byte {
-	return AppendInt64(AppendHeader(dst, bsontype.Int64, key), i64)
+	return AppendInt64(AppendHeader(dst, TypeInt64, key), i64)
 }
 
 // ReadInt64 will read an int64 from src. If there are not enough bytes it
 // will return false.
 func ReadInt64(src []byte) (int64, []byte, bool) { return readi64(src) }
 
-// AppendDecimal128 will append d128 to dst and return the extended buffer.
-func AppendDecimal128(dst []byte, d128 primitive.Decimal128) []byte {
-	high, low := d128.GetBytes()
+// AppendDecimal128 will append high and low parts of a d128 to dst and return the extended buffer.
+func AppendDecimal128(dst []byte, high, low uint64) []byte {
 	return appendu64(appendu64(dst, low), high)
 }
 
-// AppendDecimal128Element will append a BSON primitive.28 element using key and
+// AppendDecimal128Element will append high and low parts of a BSON bson.Decimal128 element using key and
 // d128 to dst and return the extended buffer.
-func AppendDecimal128Element(dst []byte, key string, d128 primitive.Decimal128) []byte {
-	return AppendDecimal128(AppendHeader(dst, bsontype.Decimal128, key), d128)
+func AppendDecimal128Element(dst []byte, key string, high, low uint64) []byte {
+	return AppendDecimal128(AppendHeader(dst, TypeDecimal128, key), high, low)
 }
 
-// ReadDecimal128 will read a primitive.Decimal128 from src. If there are not enough bytes it
+// ReadDecimal128 will read high and low parts of a bson.Decimal128 from src. If there are not enough bytes it
 // will return false.
-func ReadDecimal128(src []byte) (primitive.Decimal128, []byte, bool) {
+func ReadDecimal128(src []byte) (uint64, uint64, []byte, bool) {
```

Comment:
Optional: Consider using named returns to clarify what each of these values are.

```suggestion
func ReadDecimal128(src []byte) (high uint64, low uint64, rem []byte, ok bool) {
```